pkg/client/dtclient/config_client.go

```diff
@@ -42,6 +42,24 @@ func (d *DynatraceClient) upsertDynatraceObject(ctx context.Context, theApi api.
 			return DynatraceEntity{}, err
 		}
 
+		// The network-zone API doesn't have a POST endpoint, hence, we need to treat it as an update operation
```

Comment:
why did you move these back out after the fetching on IDs?

Reply:
because they are in the end deciding about whether an update or a create will be executed...
having that early return in another function hides that side effect entirely